docs/source/peer_ledger_snapshot.md

```diff
@@ -0,0 +1,153 @@
+
+To delete a snapshot request, simply exchange `submitrequest` with `cancelrequest`. For example:
+
+```
+peer snapshot cancelrequest -C testchannel -b 1000 --peerAddress 127.0.0.1:22509 --tlsRootCertFile tls/cert.pem
+```
+
+If you submit the `listpending` command again, the snapshot should no longer appear.
+
+### Contents of a snapshot
+
+Snapshots are stored on a peer in a snapshot folder that is further organized by channel name and the ledger height when the snapshot was taken: `{snapshotsFolder}/{channelName}/{blockHeight}`.
+
+As mentioned above, the snapshot contains four folders:
+
+* **Public state**
+  * This includes the latest state of all of the assets on the channel (for example, the current owner of a car).
+* **Private data hashes**
+  * Recall from our documentation on private data that while the actual transaction data of a private data transaction is not stored on the public ledger of the channel, hashes of the data are stored. This allows the private data to be verified against the hash, for example by an organization that is added to a private data collection.
+* **Transactions IDs**
+  * Starting with Fabric v2.0, entries in the block store index take the form of `TxID_BlkNum_TxNum`. This index can be filtered for the the transactions IDs that are added after the snapshot height by evaluating block number (`BlkNum`) portion of the index entry.
+* **Collection config history**
+  * This contains the private data collections the peer is a member of.
+
+In addition, the snapshot contains several metadata files that help the data in the snapshot to be verified:
+
+* Channel name
+* Height of the ledger at the time the snapshot was taken
+* BlockHash for the last block in the snapshot
+* Hash of the public state
+* Hash of the private data hashes
+* Hash of transaction IDs
+* Hash of collection config history
+* A representative hash of the snapshot, computed by combining all of the above
+* `BlockCommitHash` for the last block in the snapshot
+* Compression algorithm used for the data files
+
+## Joining a channel using a snapshot
+
+When joining a channel without using a snapshot, a command similar to `peer channel join --blockpath mychannel.block` is issued. When joining the peer to the channel using a snapshot, issue a command similar to:
+
+```
+peer channel joinbysnapshot --snapshotPath <path to snapshot>
+```
+
+To verify that the peer has joined the channel successfully, issue a command similar to:
+
+```
+peer channel getinfo -c <name of channel joined by snapshot>
+```
+
+Additionally, install at least one chaincode being used on the channel and issue a query. A successful return of data indicates that the peer has successfully joined using the snapshot. You can then install all of the chaincodes being used on the channel.
```

Comment:
>install at least one chaincode being used on the channel,
 
if not already installed on the peer.

Also, add one more sentence here - If the snapshot is used by a new org, and the channel contains the definitions for the chaincodes that are defined using new lifecycle, the new org will need to approve the definition of such chaincodes before, the peer can be used for invoking these chaincode.

Reply:
Is it common for a chaincode to be installed on a peer before it joins a channel?

Reply:
It may happen if the chaincode is already being used in one or more existing channels on the peer.